packages/pick-registry-for-package/package.json

```
@@ -19,8 +19,9 @@
   },
   "repository": "https://github.com/pnpm/pnpm/blob/master/packages/pick-registry-for-package",
   "scripts": {
-    "test": "pnpm run compile",
-    "lint": "eslint -c ../../eslint.json src/**/*.ts",
+    "_test": "jest",
+    "test": "pnpm run compile && pnpm run _test",
```

This can use the `pretest` script:
```suggestion
    "pretest": "pnpm run compile",
    "test": "jest",
```

---

no, this is intentional.

when all tests are running, the root script compiles all packages and then runs `pnpm run-r _test`, which does not recompile the resources again.